tools/packaging/kata-deploy/scripts/kata-deploy.sh

```diff
@@ -54,6 +76,19 @@ function delete_runtimeclasses() {
 	echo "Deleting the runtime classes"
 
 	for shim in "${shims[@]}"; do
+		if [[ "${USING_NFD}" == "true" ]]; then
+
+			# Safety check to ensure NFD worker is deployed and is running
+			[[ $(kubectl get pods -n node-feature-discovery | grep nfd-worker) ]] || die "NFD worker was not found in the cluster"
+			[[ $(kubectl get pods -n node-feature-discovery | grep nfd-worker | grep Running) ]] || die "NFD worker is not currently running"
+
+			if [[ " ${!tees_relying_on_nfd[*]} " =~ " ${shim} " ]]; then
+				tee=${tees_relying_on_nfd[${shim}]}
+				echo "As the cluster relies on NodeFeatureDiscovery, let's delete the NodeFatureRule for ${shim}"
+				kubectl delete -f /opt/kata-artifacts/nodefeaturerules/${tee}-nfd-rule.yaml
```

Comment:
Don't we need to delete in inverse order of apply? i.e. delete the runtimeclass first then the NFD rule?